src/target/riscv/riscv-013.c

```diff
@@ -1958,11 +1958,14 @@ static int examine(struct target *target)
 	if (dm013_select_hart(target, info->index) != ERROR_OK)
 		return ERROR_FAIL;
 
-	enum riscv_hart_state state;
-	if (riscv_get_hart_state(target, &state) != ERROR_OK)
+	target->state = TARGET_UNKNOWN;
+	target->debug_reason = DBG_REASON_UNDEFINED;
+
+	enum riscv_hart_state state_at_examine_start;
+	if (riscv_get_hart_state(target, &state_at_examine_start) != ERROR_OK)
 		return ERROR_FAIL;
-	bool halted = (state == RISCV_STATE_HALTED);
-	if (!halted) {
+	const bool hart_halted_at_examine_start = state_at_examine_start == RISCV_STATE_HALTED;
```

Comment:
Just a side note: This part of code seems to not handle the case when the hart is unavailable (RISCV_STATE_UNAVAILABLE) - same as the original code. 

No need to solve this in the current merge request. Just a remark for a future improvement.

Reply:
Well... actually it does. Kind of. Upon entry to examine routine we unconditionally set:

```
	target->state = TARGET_UNKNOWN;
	target->debug_reason = DBG_REASON_UNDEFINED;
```

And then the assumption is that if the hart is UNAVAILABLE - then all operations should fail (like halt_go).

Reply:
Ok, thank you for explanation. (No need to change anything at the moment.)